receiver/githubactionsreceiver/trace_event_handling.go

```diff
@@ -0,0 +1,300 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubactionsreceiver
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v61/github"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+func eventToTraces(event interface{}, config *Config, logger *zap.Logger) (ptrace.Traces, error) {
+	logger.Debug("Determining event")
+	traces := ptrace.NewTraces()
+	resourceSpans := traces.ResourceSpans().AppendEmpty()
+	scopeSpans := resourceSpans.ScopeSpans().AppendEmpty()
+
+	switch e := event.(type) {
+	case *github.WorkflowJobEvent:
+		logger.Info("Processing WorkflowJobEvent", zap.Int64("job_id", e.WorkflowJob.GetID()), zap.String("job_name", e.GetWorkflowJob().GetName()), zap.String("repo", e.GetRepo().GetFullName()))
+		jobResource := resourceSpans.Resource()
+		createResourceAttributes(jobResource, e, config, logger)
+
+		traceID, err := generateTraceID(e.GetWorkflowJob().GetRunID(), int(e.GetWorkflowJob().GetRunAttempt()))
+		if err != nil {
+			logger.Error("Failed to generate trace ID", zap.Error(err))
+			return ptrace.Traces{}, fmt.Errorf("failed to generate trace ID: %w", err)
+		}
+
+		parentSpanID := createParentSpan(scopeSpans, e.GetWorkflowJob().Steps, e.GetWorkflowJob(), traceID, logger)
+		processSteps(scopeSpans, e.GetWorkflowJob().Steps, e.GetWorkflowJob(), traceID, parentSpanID, logger)
+
+	case *github.WorkflowRunEvent:
+		logger.Info("Processing WorkflowRunEvent", zap.Int64("workflow_id", e.GetWorkflowRun().GetID()), zap.String("workflow_name", e.GetWorkflowRun().GetName()), zap.String("repo", e.GetRepo().GetFullName()))
+		runResource := resourceSpans.Resource()
+
+		traceID, err := generateTraceID(e.GetWorkflowRun().GetID(), e.GetWorkflowRun().GetRunAttempt())
+		if err != nil {
+			logger.Error("Failed to generate trace ID", zap.Error(err))
+			return ptrace.Traces{}, fmt.Errorf("failed to generate trace ID: %w", err)
+		}
+
+		createResourceAttributes(runResource, e, config, logger)
+		createRootSpan(resourceSpans, e, traceID, logger)
+
+	default:
+		logger.Error("unknown event type, dropping payload")
```

Comment:
Hey @krzko,

I started using your receiver, and I'm consistently hitting this error, although my webhook is only subscribed to the `workflow_run` and `workflow_job` events.

What do you think about adding something like the following suggestion to make debugging it a little easier?

```suggestion
		logger.Error("unknown event type, dropping payload", zap.Any("event", e))
```